scripts/mod-extractor.js

```diff
@@ -26,42 +37,34 @@ function writeToFile(file) {
     });
     var result  = new Set();
     readInterface.on("line", function(line) {
-        const match = regex.exec(line);
-        if (match) {
-            var param = match[0].slice(0, -1);
-            var str = param;
-            if (param.endsWith(",")) {
-                str = param.slice(0, -1);
-            }
-            result.add(str);
+        let index = line.indexOf(pattern);
+        while(index != -1) {
+            let modVariable = extract(line, index);
+            result.add(modVariable);
+            index = line.indexOf(pattern, index + 1);
+        }   
+        if (result.size) {
             const resultArray = [...result];
             fs.writeFile(`${directoryName}/${writePath}`, resultArray.join("\n"), (err) => {
             });
         }
     });
 }
 
-function searchFilesForString(rootDir, searchStr, extension) {
-    const files = fs.readdirSync(rootDir);
+function searchFilesForString(dir, pattern, extension) {
+    const files = fs.readdirSync(dir);
     for(let i = 0; i < files.length; i++) {
-        const dir = path.join(rootDir, files[i]);
-        if (!fs.statSync(dir).isDirectory()) {
-            continue;
-        }
-        fs.readdir(dir, function(err, files) {
-            if (err) {
-                console.warn(err);
-                return;
+        const file = path.join(dir, files[i]);
+
+        if (fs.statSync(file).isDirectory()) {
+            searchFilesForString(file, pattern, extension);
+        } else {
+            if (path.extname(file) === extension) {
+                writeToFile(file);
             }
-            files.forEach(function (file) {
-                let readFile = path.join(dir, file);
-                if (path.extname(readFile) === extension) {
-                    writeToFile(readFile);
-                } 
-            });
-        });
+        }
     }
 }
 
-searchFilesForString(rootDir, regex, fileExtension);
+searchFilesForString(rootDir, pattern, fileExtension);
```

Comment:
How will you execute this script during the build process?

Reply:
Currently, this script is not part of the build process. User can trigger the build with the following command
yarn docs:mod

Reply:
Okay Nice!!! But this also needs to go into the build process eventually. Also please think on that part on how you can include this inside the component-builder build process.

Reply:
@Rajdeepc True! Patrick & I had talked about making this manual for now until we can merge in your updates that way there is less likely to be a conflict.